authorization/v2/uniformResourceIdentifier.go

```diff
@@ -24,6 +24,64 @@ func NewUriFromRequest(resource resources.Resource, handler gin.HandlerFunc, ctx
 	}
 }
 
+// NewURIFromString parses the string representation of a URI into the UniformResourceIdentifier struct.
+// NewURIFromString expects the string to be of the following form, otherwise ErrInvalidURI is returned.
+// hs:<service_name>:<subsystem>:<version>:<category>:<resource_name>?<allowed_arguments>#<permission_metadata>
+func NewURIFromString(source string) (UniformResourceIdentifier, error) {
```

Comment:
It would be great if this function unescaped the arguments and the metadata with https://golang.org/pkg/net/url/#QueryUnescape

Reply:
Yep, on the todo list 😄

Reply:
Ok nice, just wanted to make sure :slightly_smiling_face: